python/ray/_raylet.pyx

```diff
@@ -352,8 +352,12 @@ cdef class RayletClient:
 
     def prepare_actor_checkpoint(self, ActorID actor_id):
         cdef CActorCheckpointID checkpoint_id
-        check_status(self.client.get().PrepareActorCheckpoint(
-            actor_id.data, checkpoint_id))
+        cdef CActorID c_actor_id = actor_id.data
+        # PrepareActorCheckpoint will wait for raylet's reply, release
+        # the GIL so other Python threads can run.
+        with nogil:
+            check_status(self.client.get().PrepareActorCheckpoint(
```

Comment:
Is `PrepareActorCheckpoint` a blocking call, which waits for a response from the raylet? If yes, then we definitely need to release the GIL. If not, then we could still release the GIL, though in that case there are many other methods like this where we should release the GIL.

Reply:
Ok, I just checked and see that it waits for a response.